### ISSUE TITLE
Security vulnerability found in dot-prop

### ISSUE THREAD

**Issue by lukasholzer** [conventional-changelog/conventional-changelog#592]
# Security vulnerability found

Vulnerable module: dot-prop
Introduced through: compare-func@1.3.2

## Detailed paths
* **Introduced through**: conventional-changelog-angular@5.0.6 › compare-func@1.3.2 › dot-prop@3.0.0

https://app.snyk.io/test/npm/conventional-changelog-angular/5.0.6

## Overview

[dot-prop](https://github.com/sindresorhus/dot-prop#readme) is a package to get, set, or delete a property from a nested object using a dot path.

Affected versions of this package are vulnerable to Prototype Pollution. It is possible for a user to modify the prototype of a base object.

### PoC by aaron_costello

```javascript
var dotProp = require("dot-prop")
const object = {};
console.log("Before " + object.b); //Undefined
dotProp.set(object, '__proto__.b', true);
console.log("After " + {}.b); //true
```

### Remediation

Upgrade dot-prop to version 5.1.1 or higher.

### References

[GitHub Commit
](https://github.com/sindresorhus/dot-prop/commit/3039c8c07f6fdaa8b595ec869ae0895686a7a0f2)
[HackerOne Report](https://hackerone.com/reports/719856)

**Comment by adamweeks** [comment]
Thank you for opening this issue, there is also a NVD alert on this package as well:

https://nvd.nist.gov/vuln/detail/CVE-2020-8116

**Comment by iamcdonald** [comment]
Related to the above,
I've raised a couple of PR's on `compare-func`

https://github.com/stevemao/compare-func/pull/4 - fix the tests as they're currently failing
https://github.com/stevemao/compare-func/pull/3 - update the `dot-prop` version to > 5.1.0

**Comment by bcoe** [comment]
👋 I just updated a variety of dependencies, is this still an issue?

**Comment by viveksharma2175** [comment]
I am still getting Prototype Pollution vulnerability for dot-prop through "conventional-changelog-conventionalcommits": "4.3.0" used in @commitlint/config-conventional 9.1.2

![image](https://user-images.githubusercontent.com/48869892/87906186-82bc1400-caa5-11ea-8d28-69e2674ed434.png)

**Comment by derykmarl** [comment]
Same.  It's coming up as a HIGH severity vulnerability now.

```
                       === npm audit security report ===


                                 Manual Review

             Some vulnerabilities require your attention to resolve

          Visit https://go.npm.me/audit-guide for additional guidance
High            Prototype Pollution

  Package         dot-prop


  Patched in      >=5.1.1

  Dependency of   gatsby

  Path            gatsby > devcert > configstore > dot-prop

  More info       https://npmjs.com/advisories/1213

  High            Prototype Pollution

  Package         dot-prop

  Patched in      >=5.1.1

  Dependency of   gatsby

  Path            gatsby > gatsby-cli > update-notifier > configstore >
                  dot-prop

  More info       https://npmjs.com/advisories/1213
```

How do we fix this?

**Comment by danmana** [comment]
Fixed by PR #647 
Can be closed when conventional-changelog-angular 5.0.11 is released. See #658

**Comment by bcoe** [comment]
@lukasholzer I believe this should be fixed, let me know if it continues to be an issue.
